src/main/java/com/tidyjava/bp/PostReader.java

```diff
@@ -0,0 +1,65 @@
+package com.tidyjava.bp;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static java.util.Collections.emptyList;
+import static java.util.stream.Collectors.toList;
+
+@Service
+public class PostReader {
+
+    @Value("${posts.location}")
+    private String postsLocation;
+
+    public List<MarkdownPost> readAll() {
+        URL postsRes = getResource(postsLocation);
+        if (notExists(postsRes)) {
+            return emptyList();
+        }
+        File postsDir = toFile(postsRes);
```

Comment:
The logic from 25-28 doesn't need to be executed with every request, it won't change during service lifecycle